server/src/main/java/org/elasticsearch/index/mapper/DocumentParserContext.java

```diff
@@ -336,7 +344,25 @@ public final void addDynamicMapper(Mapper mapper) {
         // dynamically mapped objects when the incoming document defines no sub-fields in them:
         // 1) by default, they would be empty containers in the mappings, is it then important to map them?
         // 2) they can be the result of applying a dynamic template which may define sub-fields or set dynamic, enabled or subobjects.
-        dynamicMappers.computeIfAbsent(mapper.name(), k -> new ArrayList<>()).add(mapper);
+        dynamicMappers.computeIfAbsent(mapper.name(), k -> new ArrayList<>(1)).add(mapper);
+        return true;
+    }
+
+    /*
+     * Returns an approximation of the number of dynamically mapped fields that will be added to the mapping.
```

Comment:
Would it be possible to clarify why it is an approximation and mention situations where the count is not accurate? I am trying to understand the end-user impact for this and I think future readers could benefit from additional info.